test/e2e/tests/test_queue.py

```diff
@@ -137,3 +139,52 @@ def test_crud(self, simple_queue):
         tags.assert_equal_without_ack_tags(
             expect_after_update_tags, latest_tags,
         )
+
+    def test_adoptedresource(self):
+        queue_name = random_suffix_name("sqs-queue-acktest", 24)
+        queue_url = sqsqueue.create_queue(queue_name)
+
+        sqsqueue.wait_until_exists(queue_name)
```

Comment:
Creating and deleting test requirements inside the test function is a bit dangerous because if an assertion fails, then the cleanup never occurs. Could you please move the logic for the `create_queue` and eventual `wait_until_deleted` into a fixture (you can `yield` the name) so that we are guaranteed to have proper cleanup.

Reply:
[refactor e2e test with adoption test helper](https://github.com/aws-controllers-k8s/sqs-controller/pull/32/commits/0ffd66f03e9297bf5bb1dd7a9256fefc61fd774d)

Reply:
`adoption.AbstractAdoptionTest` is really helpful!